### DISCUSSION TITLE
Coral. Synology. DSM7.7 (DS918+)

### DISCUSSION THREAD

**Discussion by shatreyman** [blakeblackshear/frigate#9009] [Ask A Question]
Hi! There are some threads about same question, but no good solution. Might be this one will lead to solution.
I'm running HomeAssistace on Synology in Virtual Machine. Recently recieved Coral device which is identified on PC as a Coral USB accelerator from Google (after installation of drivers). Coral is plugged thorough powered USB hub 3.1 https://www.unitek-products.com/products/usb3-0-4-port-hub
![image](https://github.com/blakeblackshear/frigate/assets/78681753/1d81ee0c-7711-4730-acb5-0e71f29e06c9)

Have not tested coral work on PC.

On a Synology i tried to connect it directly to USB port, thorough powered HUB, unpowered HUB and even powered HUB connected to PC and then without pulling out power supply to Synology. The result is always same "USBdevice - Global Unichip Corp."
![image](https://github.com/blakeblackshear/frigate/assets/78681753/8fd2c865-78e0-44ae-ac03-2573c6979a20)

(CP210x... is SkyConnect stick which is working in any setup).
lsusb command result in terminal:
![image](https://github.com/blakeblackshear/frigate/assets/78681753/59508c7c-3d89-4ddc-b55e-ab2152b9e4d2)

Virtual Machine:
![image](https://github.com/blakeblackshear/frigate/assets/78681753/db4c74cb-931d-4ee4-9268-84dcec616289)

HomeAssistant hardware view:
![image](https://github.com/blakeblackshear/frigate/assets/78681753/16bb62d0-10b3-4279-b6b8-0fab267f595c)

In config file configuration must be correct:
detectors:
  coral:
    type: edgetpu
    device: usb

So, in my opinion there are several possibilities:

1. Coral not possible to run on Synology at all because of some missing drivers or something like that.
2. Coral requires other USB hub for some reason. There is information in web that TP-LINK hub may help.
3. There is something in HomeAssistance must be changed or in Frigate to provide information that Coral is on /dev/bus/usb/003/002 or that its name is something else.

Any suggestions?

**Comment by shatreyman** [comment]
bump

**Reply by alrassia** [reply]
I am running a coral usb on synology as well so it does work. Indeed synology is shipped by default without the necessary usb drivers for serial devices in the most recent versions. You will have to install them yourself first before it will work.

check out this repository for details: https://github.com/robertklep/dsm7-usb-serial-drivers

**Reply by shatreyman** [reply]
Unfortunately nothing changed :( Used for guidelines beradedTinker video https://www.youtube.com/watch?v=GjZI4Mw0a4c&t=624s but nope, still "USBdevice - Global Unichip Corp"

What Synology you have? How Coral is connected to USB, directly or through HUB?

**Reply by alrassia** [reply]
Ds923+ with 7.2.1 running, i use the r1000 modules from the website. My coral is directly connected to one of the 2 usb ports on my synology (front to be precise). Which modules to use depends on your cpu architecture, i have a ryzen cpu in mine.

**Reply by shatreyman** [reply]
Hmm... I'll keep trying.
I used Apollolake modules as written on manual (for DS918+). I have one difference from original Synology, I have added 4GB mode RAM, but it should not be an issue.

**Reply by alrassia** [reply]
What is your docker configuration like? How are you passing the usb to the container?

**Reply by shatreyman** [reply]
Im running HASS on virtual machine, not using docker.

**Reply by alrassia** [reply]
How are you running frigate then?

**Reply by shatreyman** [reply]
It is integrated to HomeAssistance as an Add-on, https://docs.frigate.video/integrations/home-assistant/

**Reply by alrassia** [reply]
Does frigate see and use the coral? I checked on my synology and it also does not report as a coral but works without issue.

**Reply by shatreyman** [reply]
Frigate cant be started, or it is starting but failing as it cant find Coral. With config:

detectors:
  cpu1:
    type: cpu

it is working, but slowly, object recognition isnt good.

Might be Frigate needs to be slightly chaged, so it searches not "google device", but it is enough for "global unichip". Might be coral is working, but Frigate do not know what the device name is.

**Reply by alrassia** [reply]
So i checked my install, I run Frigate as a docker image. However on the synology as well as in the frigate docker image the coral is identified the same as how you see it. So the issue is likely not there.

There is a known behaviour that the coral changes its usb id after it runs, the workaround is to pass the port to the VM instead of the device id. See for example https://www.reddit.com/r/Proxmox/comments/nmsknx/proxmox_vm_ubuntu_2004_connect_google_coral_usb/
The solution would be to pass the usb port to the VM instead of the specific device id.
